scrapy_poet/overrides.py

```diff
@@ -1,19 +1,17 @@
 import logging
 from abc import ABC, abstractmethod
 from collections import defaultdict
-from typing import Callable, Dict, Iterable, List, Mapping, Optional, Tuple, Type, Union
+from typing import Any, Callable, Dict, Iterable, Mapping, Optional, Type
+from warnings import warn
 
 from scrapy import Request
 from scrapy.crawler import Crawler
-from url_matcher import Patterns, URLMatcher
-from web_poet import ItemPage
+from url_matcher import URLMatcher
+from web_poet import ItemPage, RulesRegistry
 from web_poet.rules import ApplyRule
 
 logger = logging.getLogger(__name__)
 
-RuleAsTuple = Union[Tuple[str, Type[ItemPage], Type[ItemPage]], List]
-RuleFromUser = Union[RuleAsTuple, ApplyRule]
-
 
 class OverridesRegistryBase(ABC):
```

Comment:
Do we still need this ABC? I think we should either remove it, or update to include all the methods which users need to implement in order for registry to work. Removing might be easier.

Reply:
I don't think we'll need it anymore.

I'm planning to remove it in a succeeding PR after I submit another PR which moves most of the registry functionalities to `web-poet`. The `web_poet.rules.Registry` should implement `overrides_for()` and `page_objects_for()` from the start.

I also think the new `registry.py` in https://github.com/scrapinghub/scrapy-poet/pull/103 would shrink a lot.